crates/revm/src/evm_impl.rs

```diff
@@ -185,6 +185,7 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> Transact
             match exit_reason {
                 return_ok!() => {
                     gas.erase_cost(ret_gas.remaining());
+                    #[cfg(not(feature = "optional_gas_refund"))]
```

Comment:
You introduced a config flag, but you are not using it :D
And I think a better place for this is where the refund calculation is done in the `finalize` function.

Reply:
+ actually using cfg flag 😅 
+ moved to finalize